src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -41,13 +43,74 @@ namespace ts.server.typingsInstaller {
         onRequestCompleted: RequestCompletedAction;
     }
 
+    interface TypeDeclarationTimestampFile {
+        entries: MapLike<number>;
+    }
+
+    function loadTypeDeclarationTimestampFile(typeDeclarationTimestampFilePath: string, host: InstallTypingHost, log: Log): MapLike<number> {
+        const fileExists = host.fileExists(typeDeclarationTimestampFilePath);
+        if (!fileExists) {
+            if (log.isEnabled()) {
+                log.writeLine(`Type declaration timestamp file '${typeDeclarationTimestampFilePath}' does not exist`);
+            }
+        }
+        try {
+            if (fileExists) {
+                const content = <TypeDeclarationTimestampFile>JSON.parse(host.readFile(typeDeclarationTimestampFilePath));

```

Comment:
Our `readFile` method returns `undefined` if the file doesn't exist, so you don't need the above call to `fileExists`

Reply:
Refactored.